### PR TITLE
Refactoring the System Proxy

### PR DESCRIPTION
**Pull request by DuckSoft**
- refactored linux system proxy setting routine
 - added hint and damn workaround for the deepin desktop environment

### CODE REVIEW

**Comment by DuckSoft** [comment]
@Qv2ray/management this pull request is ready to be merged into `dev` branch.
additional translations will be immediately ready for merging.

**Comment by DuckSoft** [comment]
Translation Ready. Just have this checked and merge it~

File: src/components/proxy/QvProxyConfigurator.cpp [hunk 1]
```diff
@@ -232,61 +232,101 @@ namespace Qv2ray::components::proxy
         QStringList actions;
         actions << QString("gsettings set org.gnome.system.proxy mode '%1'").arg("manual");
         bool isKDE = qEnvironmentVariable("XDG_SESSION_DESKTOP") == "KDE";
+        bool isDDE = isKDE ? false : qEnvironmentVariable("XDG_SESSION_DESKTOP") == "DDE";
```
Comment (ymshenyu) [inline src/components/proxy/QvProxyConfigurator.cpp:235]:
```suggestion
        bool isDDE = isKDE ? false : qEnvironmentVariable("XDG_SESSION_DESKTOP") == "deepin";
```
`XDG_SESSION_DESKTOP` is equal to `deepin` on DDE. You should consider to change it.

**Comment by Qv2ray-dev** [comment]
blocked by a requested change @DuckSoft

**Pull request merged**